python/src/aac/io/parser/_yaml.py

```diff
@@ -31,11 +65,31 @@ def parse_yaml(source: str, content: str) -> list[dict]:
         models = list(load_all(content, Loader=SafeLoader))
         _error_if_not_yaml(source, content, models)
         _error_if_not_complete(source, content, models)
-        return models
     except YAMLParserError as error:
-        raise ParserError(source, [f"Failed to parse file, invalid YAML {error.context} {error.problem}", content])
+        logging.error(f"Encountered YAML Parsing Error-Problem: {error.problem}. Encountered in: {source} at {error.context}")
+        if error.problem_mark:
+            logging.error(f"Error occurs at line, column: {error.problem_mark.line+1}, {error.problem_mark.column+1}")
+        logging.error(f"Content of error: {content}")
+
+        raise ParserError(source, ["Encountered an invalid YAML with the following parser error: {error.problem}",
+                                   f"Encountered error at line, column: {error.problem_mark.line+1}, {error.problem_mark.column+1}",
+                                   f"Context of the error: {error.context}"]) from None
+    except YAMLScannerError as error:
+        logging.error(f"Encountered YAML Scanner Error-Problem: {error.problem}. Encountered in: {source} at {error.context}")
+        if error.problem_mark:
+            logging.error(f"Error occurs at line, column: {error.problem_mark.line+1}, {error.problem_mark.column+1}")
+        logging.error(f"Content of error: {content}")
+
+        raise ParserError(source, [f"Encountered an invalid YAML with the following scanner error: {error.problem}",
+                                   f"Encountered error at line, column: {error.problem_mark.line+1}, {error.problem_mark.column+1}",
+                                   f"Context of the error: {error.context}"]) from None
     except Exception as error:
-        raise ParserError(source, [f"Failed to parse file, invalid YAML {error}", content])
+        logging.error(f"Error: {error}. Encountered in: {source}")
+        logging.error(f"Content of error: {content}")
+
+        raise ParserError(source, [f"Encountered the following error: {error}"]) from None
```

Comment:
Since we're re-using these error messages in the same file, let's extract them out to some private helper functions inside this module.

Reply:
I pulled out the logging and parser error pieces to private helper functions.